### PR TITLE
Add Azure pipeline definition

### PR DESCRIPTION
**Pull request by ecraig12345**
~~Add a build pipeline definition for moving the build from CircleCI to Azure Devops. (Migration plan is probably to run both for a few days, then delete CircleCI if all goes well.)~~ EDIT: We discovered the ADO machines are slower than the CircleCI ones and take about 1.5-2x as long per task, so we're sticking with CircleCI for now. The Azure build will only run for component governance checks on master.

Also switch Screener and CircleCI to use Node 10.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluent-ui-react/pull/2125)

### CODE REVIEW

**Comment by ecraig12345** [comment]
Main build tasks were successful, but I'm marking this as do not merge until all the API keys and such have been added to the variable group (and I have to check the spelling of the branch name variables).

**Review by kenotron — APPROVED** [review]

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      fi
+    displayName: Bundle statistics (master only)
+
+  - script: |
+      if [ $BUILD_SOURCEBRANCHNAME == 'master' ]; then yarn perf; fi
+    displayName: Performance tests (master only)
+
+  - script: |
+      if [ $BUILD_SOURCEBRANCHNAME == 'master' ] && [ -n "${STATS_URI}" ]; then
+        yarn stats:save --tag=`git tag --points-at HEAD`
```
Comment (miroslavstastny) [inline azure-pipelines.yml]: `stats:save` reads several env variables, not sure all are available under the same name in azure.
See https://github.com/microsoft/fluent-ui-react/blob/master/build/gulp/tasks/stats.ts#L158-L167

Comment (kenotron) [reply azure-pipelines.yml]: Great call out... we'll have to map those to these: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables

Comment (ecraig12345) [reply azure-pipelines.yml]: Thanks for catching that! I'll update it to check both sets of variables for now.

Comment (ecraig12345) [reply azure-pipelines.yml]: Should be fixed now

**Review by kenotron — CHANGES_REQUESTED** [review]
offline convo: will only use ADO for publishes and compliance.

**Review by kenotron — APPROVED** [review]
GO!

**Pull request merged**